packages/components/src/Typography/Typography.tsx

```diff
@@ -51,18 +53,27 @@ export function Typography({
   textCase,
   textColor,
   emphasisType,
+  numberOfLines,
 }: TypographyProps) {
+  const shouldTruncateText = numberOfLines && numberOfLines > 0;
   const className = classnames(
     styles.base,
     fontWeights[fontWeight],
     size && fontSizes[size],
     textCase && textCases[textCase],
     textColor && textColors[textColor],
     emphasisType && emphasis[emphasisType],
+    shouldTruncateText && truncate.textTruncate,
   );
 
+  let truncateLines: CSSProperties | undefined;
+  if (shouldTruncateText) {
+    truncateLines = {};
+    truncateLines.WebkitLineClamp = numberOfLines;
+  }
+
   return (
-    <Tag id={id} className={className}>
+    <Tag id={id} className={className} style={truncateLines}>
```

Comment:
debating whether or not I think it would be better to have a handful of css classes defined

eg.

```
.truncate--small {}

.truncate--large {}
```

which lets us stick to only `className` and avoid adding an inline style

vs this approach that simply takes the number of lines we want to limit it to